src/vs/workbench/browser/web.main.ts

```diff
@@ -118,26 +120,39 @@ export class BrowserMain extends Disposable {
 			const timerService = accessor.get(ITimerService);
 			const openerService = accessor.get(IOpenerService);
 			const productService = accessor.get(IProductService);
+			const outputService = accessor.get(IOutputService);
 			const telemetryService = accessor.get(ITelemetryService);
 			const progessService = accessor.get(IProgressService);
 
+			let outputChannelCounter = 0;
+
 			return {
 				commands: {
 					executeCommand: (command, ...args) => commandService.executeCommand(command, ...args)
 				},
 				env: {
 					uriScheme: productService.urlProtocol,
+					telemetryLevel: telemetryService.telemetryLevel,
 					async retrievePerformanceMarks() {
 						await timerService.whenReady();
 
 						return timerService.getPerformanceMarks();
 					},
 					async openUri(uri: URI): Promise<boolean> {
 						return openerService.open(uri, {});
-					},
-					telemetryLevel: telemetryService.telemetryLevel,
+					}
 				},
 				window: {
+					createOutputChannel: async (name, languageId) => {
+						const id = `web-embedder-${++outputChannelCounter}`;
```

Comment:
@sandy081 can you maybe check this code for whether that is a good use of the output channel API?